app/services/cache/mapping/declaration/DeclarationBuilder.scala

```diff
@@ -0,0 +1,91 @@
+
+  def build(exportsCacheModel: ExportsCacheModel): Declaration = {
+    val declaration = new Declaration()
+
+    declaration.setFunctionCode(FunctionCodeBuilder.build(defaultFunctionCode))
+    declaration.setFunctionalReferenceID(createDeclarationFunctionalReferenceIDType(exportsCacheModel))
+
+    val typeCode: Option[DeclarationTypeCodeType] = exportsCacheModel.additionalDeclarationType.map(
+      additionalDeclarationType =>
+        TypeCodeBuilder.createTypeCode(additionalDeclarationType, exportsCacheModel.dispatchLocation)
+    )
+
+    declaration.setTypeCode(typeCode.getOrElse(new DeclarationTypeCodeType()))
+    declaration.setGoodsItemQuantity(GoodsItemQuantityBuilder.createGoodsItemQuantity(exportsCacheModel.items.toSeq))
+
+    generateAgent(exportsCacheModel).foreach { data =>
+      declaration.setAgent(data)
+    }
+
+    //    declaration.setGoodsShipment(GoodsShipmentBuilder.build)
+    //    declaration.setExitOffice(ExitOfficeBuilder.build)
+    //    declaration.setBorderTransportMeans(BorderTransportMeansBuilder.build)
+    //    declaration.setExporter(ExporterBuilder.build)
+    //    declaration.setDeclarant(DeclarantBuilder.build)
+    //    declaration.setInvoiceAmount(InvoiceAmountBuilder.build)
+    //    declaration.setPresentationOffice(PresentationOfficeBuilder.build)
+    //    declaration.setSpecificCircumstancesCodeCode(SpecificCircumstancesCodeBuilder.build)
+    //    declaration.setSupervisingOffice(SupervisingOfficeBuilder.build)
+    //    declaration.setTotalPackageQuantity(TotalPackageQuantityBuilder.build)
+    //    declaration.setConsignment(DeclarationConsignmentBuilder.build)
+    //    declaration.setTypeCode(TypeCodeBuilder.build)
+    //
+    //    val authorisationHolders = AuthorisationHoldersBuilder.build
+    //    if (authorisationHolders != null && !authorisationHolders.isEmpty) {
+    //      declaration.getAuthorisationHolder.addAll(authorisationHolders)
+    //    }
+    //
+    //    val currencyExchangeList = CurrencyExchangeBuilder.build
+    //    if (currencyExchangeList != null && !currencyExchangeList.isEmpty) {
+    //      declaration.getCurrencyExchange.addAll(currencyExchangeList)
+    //    }
+
+    declaration
+  }
+
+  private def createDeclarationFunctionalReferenceIDType(
+    exportsCacheModel: ExportsCacheModel
+  ): DeclarationFunctionalReferenceIDType = {
+    val returnVal = new DeclarationFunctionalReferenceIDType()
+    exportsCacheModel.consignmentReferences.map(
+      consignmentReferences => if (consignmentReferences.lrn.nonEmpty) returnVal.setValue(consignmentReferences.lrn)
+    )
+    returnVal
+  }
+
+  private def generateAgent(exportsCacheModel: ExportsCacheModel): Option[Declaration.Agent] =
+    exportsCacheModel.parties.representativeDetails.map { representativeDetails =>
+      if (RepresentativeDetails.isDefined(representativeDetails)) {
+        AgentBuilder.createAgent(representativeDetails)
+      } else {
+        new Declaration.Agent()
```

Comment:
This should not happen. You need to remove so.
Also, I still think it is better to delegate the creation of DMS declaration subtypes to its own classes, otherwise the declaration builder will be a very long class and quite bloated